sql-plugin-api/pom.xml

```diff
@@ -57,6 +57,29 @@
                 </dependency>
             </dependencies>
         </profile>
+        <profile>
+            <!--
+                 Note that we are using the Spark version for all of the Databricks dependencies as well.
```

Comment:
This comment scared me a bit, because we definitely do _not_ want to use Spark 3.3.2 dependencies for the spark332db profile, for example.  Would be nice to clarify this a bit (e.g.: "custom Databricks Spark version" instead of "Spark version", otherwise it implies we could be trying to pull public dependencies (or try to alias the Databricks jars as if they were public dependencies) which is asking for trouble if that's the case.

Also not sure how this change relates to the purpose of this PR.  Seems like a separate change.

Reply:
I had to duplicate some old code with the code we had for a long while https://github.com/NVIDIA/spark-rapids/blame/ba28548cceb391c3341c68bdbb31da4ab8c28ab7/pom.xml#L310-L316. We can improve the comment of course.